Sources/FaceLiveness/FaceDetection/BlazeFace/DetectedFace.swift

```diff
@@ -14,53 +14,78 @@ struct DetectedFace {
     let rightEye: CGPoint
     let nose: CGPoint
     let mouth: CGPoint
+    let rightEar: CGPoint
+    let leftEar: CGPoint
 
     let confidence: Float
 
-    func boundingBoxFromLandmarks() -> CGRect {
-        let eyeCenterX = (leftEye.x + rightEye.x) / 2
-        let eyeCenterY = (leftEye.y + rightEye.y) / 2
-
-        let cx = (nose.x + eyeCenterX) / 2
-        let cy = (nose.y + eyeCenterY) / 2
-
-        let ow = sqrt(pow((leftEye.x - rightEye.x), 2) + pow((leftEye.y - rightEye.y), 2)) * 2
-        let oh = 1.618 * ow
-        let minX = cx - ow / 2
-        let minY = cy - oh / 2
-
-        let rect = CGRect(x: minX, y: minY, width: ow, height: oh)
+    func boundingBoxFromLandmarks(ovalRect: CGRect) -> CGRect {
+        let alpha = 2.0
+        let gamma = 1.8
+        let ow = (alpha * pupilDistance + gamma * faceHeight) / 2
+        var cx = (eyeCenterX + nose.x) / 2
+        
+        if ovalRect != CGRect.zero {
+            let ovalTop = ovalRect.minY
+            let ovalHeight = ovalRect.maxY - ovalRect.minY
+            if eyeCenterY > (ovalTop + ovalHeight) / 2 {
+                cx = eyeCenterX
+            }
+        }
+        
+        let faceWidth = ow
+        let faceHeight = 1.68 * faceWidth
+        let faceBoxBottom = boundingBox.maxY
+        let faceBoxTop = faceBoxBottom - faceHeight
+        let faceBoxLeft = min(cx - ow / 2, rightEar.x)
+        let faceBoxRight = min(cx + ow / 2, leftEar.x)
+        
+        let rect = CGRect(x: faceBoxLeft, y: faceBoxTop, width: faceBoxRight, height: faceBoxBottom)
         return rect
     }
 
     var faceDistance: CGFloat {
         sqrt(pow(rightEye.x - leftEye.x, 2) + pow(rightEye.y - leftEye.y, 2))
     }
+    
+    var pupilDistance: CGFloat {
```

Comment:
looks like this is the same value as the `faceDistance` variable above?

Reply:
yes, it equates to the same value, but I'm leaving it as is to match the document and readability/clarity since `faceDistance` is currently used to display oval fit instruction and `pupilDistance` is used to calculate the bounding box

Reply:
fwiw that calculation is really "face distance based on pupil distance", which I think is where the confusing / double naming comes in.